architecture/additional_concepts/dynamic_admission_controllers.adoc

```diff
@@ -0,0 +1,473 @@
+kind: Deployment
+metadata:
+  annotations:
+    "initializer.kubernetes.io": "true"
+  labels:
+    app: helloworld
+  name: helloworld
+...
+----
+
+[[architecture-additional-concepts-dynamic-admission-initializer-config]]
+=== Configure Initializers
+
+you should first deploy the initializer controller and make sure that it is
+working properly before creating the `initializerConfiguration`. Otherwise, any
+newly created resources will be stuck in an uninitialized state.
+
+The following list represents a set of best practices to follow when building Initializers.
+
+* Initializers must have a unique fully qualified name. Examples: initializer.vaultproject.io initializer.example.com
+* Initializers should be deployed using a Deployment for easy upgrades and auto restarts.
+* Initializers should explicitly set the list of pending initializers to exclude itself, or to an empty array, to avoid getting stuck waiting to initialize. Examples:
+
+. Configure which initializers are enabled and which resource types need initialization
+by creating *initializer configuration* resources.
++
+The following is an example `initializerConfiguration`:
++
+[source-yaml]
+----
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: InitializerConfiguration
+metadata:
+  name: example-config
+initializers:
+  # the name needs to be fully qualified, i.e., containing at least two "."
+  - name: podimage.example.com
+    rules:
+      # apiGroups, apiVersion, resources all support wildcard "*".
+      # "*" cannot be mixed with non-wildcard.
+      - apiGroups:
+          - ""
+        apiVersions:
+          - v1
+        resources:
+          - pods
+----
++
+Ensure that all expansions of the `<apiGroup, apiVersions, resources>` tuple
+in a `rule` are valid. If they are not, separate them in different `rules`.
++
+After you create the `initializerConfiguration`, the system will take a few
+seconds to honor the new configuration. 
+
+. {product-title} appends the name of the initializer configuration object, here  `podimage.example.com`, 
+to the `metadata.initializers.pending` field of any newly created pods. 
+
+. An initializer controller should list and watch for uninitialized objects, by
+using the query parameter `includeUninitialized=true`. If using client-go, just
+set 
+[listOptions.includeUninitialized](https://github.com/kubernetes/kubernetes/blob/v1.7.0-rc.1/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L315)
+to true.
+
+////
+
+[[architecture-additional-concepts-dynamic-admission-webhooks]]
+== External Admission Webhooks
+
+External admission webhooks are HTTP callbacks that receive
+admission requests and do something with them. What an external admission
+webhook does is up to you, but there is an
+link:https://github.com/kubernetes/kubernetes/blob/v1.7.0-rc.1/pkg/apis/admission/v1alpha1/types.go[interface]
+that it must adhere to so that it responds with whether or not the
+admission request should be allowed. 
+
```

Comment:
Do we want the above link?

Reply:
The current types are at https://github.com/kubernetes/kubernetes/blob/v1.10.0-beta.1/staging/src/k8s.io/api/admission/v1beta1/types.go